### PR TITLE
Upgrade from Go 1.13.4 -> 1.14.2

### PR DESCRIPTION
**Pull request by y0ssar1an**
This is a straightforward compiler version bump. All the tests passed with no issues. It's nice to finally delete this boilerplate.

```
GO111MODULE=on
GOFLAGS=-mod=vendor
```

Note that because it touches `vendor/modules.txt` there will be a merge conflict with #6219. It's easy to fix. When that merges, I'll simply rebase to delete `vendor/` entirely.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -24,4 +24,4 @@ require (
 	google.golang.org/api v0.22.0
 )
 
-go 1.13
+go 1.14
```
Comment (y0ssar1an) [inline go.mod:27]: Bumping this to 1.14 does not prevent it from building with Go 1.13. Go 1.13.9+ will build it just fine.

The behavior of Go modules is still being tweaked with every compiler release. This directive simply instructs the compiler to use the latest behavior. It's a good idea to keep it up-to-date.

File: vendor/modules.txt [hunk 2]
```diff
@@ -9,6 +9,7 @@ cloud.google.com/go/internal/version
 cloud.google.com/go/longrunning
 cloud.google.com/go/longrunning/autogen
 # cloud.google.com/go/bigtable v1.1.0
+## explicit
```
Comment (y0ssar1an) [inline vendor/modules.txt:12]: `go mod vendor` added this 🤷‍♂️

**Comment by danawillow** [comment]
Handing off to @megan07 since it's super relevant to the vendor removal work

**Comment by y0ssar1an** [comment]
This PR has fallen behind `master`. Apparently, [we're going to keep `vendor/` after all](https://github.com/terraform-providers/terraform-provider-google/pull/6356). That's fine. I will resubmit this with some additional changes that reflect the new direction.

**Comment by danawillow** [comment]
Hey @y0ssar1an, there's a lot of things underway. That revert was due to a short test of the vendor removal changes. Please be patient :)

**Pull request closed without merging**